hierarchicalforecast/methods.py

```diff
@@ -966,6 +970,35 @@ def fit_predict(self,
 
     __call__ = fit_predict
 
+@njit(parallel=True, fastmath=True, error_model="numpy")
+def _ma_cov(residuals: np.ndarray, not_nan_mask: np.ndarray):
+    """Masked empirical covariance matrix.
+
+    :meta private:
+    """
+    n_timeseries = residuals.shape[0]
+    W = np.zeros((n_timeseries, n_timeseries), dtype=np.float64).T
```

Comment:
```suggestion
    W = np.empty((n_timeseries, n_timeseries), dtype=np.float64)
```

It will be a tad faster to not initialise the values in the array as we are setting them all in the loop, plus no need for the transpose.

Reply:
Thanks - did you test this? I wrote it like this because benchmarks showed the zero initialization + transpose is faster later on (it has to do with the memory lay out being more convenient for later use).

Reply:
Ran some benchmarks - only `nogil=True` seems to have an effect, so I'm making that change.

Base case:

![image](https://github.com/user-attachments/assets/d1290d6e-ab6b-41e3-b1bc-09a86572bce1)

Not transposing W on init:

![image](https://github.com/user-attachments/assets/ee463753-44d1-4f23-ac1d-9658669aeb54)

Static typing + nogil + cache:

![image](https://github.com/user-attachments/assets/d1431ce5-c8c4-4dfe-a0f0-cda3ea5008ac)

Nogil:

![image](https://github.com/user-attachments/assets/64e9d19c-3c43-4c8a-88e8-6264b92397d5)

Reply:
You can initialize it with fortran layout:
```python
np.empty((n_timeseries, n_timeseries), dtype=np.float64, order='F')
```

`empty` should be faster than `zeros`